providers/implementations/ciphers/ciphercommon_block.c

```diff
@@ -120,18 +128,19 @@ int ossl_cipher_unpadblock(unsigned char *buf, size_t *buflen, size_t blocksize)
      * Otherwise it provides a padding oracle.
      */
     pad = buf[blocksize - 1];
-    if (pad == 0 || pad > blocksize) {
-        ERR_raise(ERR_LIB_PROV, PROV_R_BAD_DECRYPT);
-        return 0;
-    }
-    for (i = 0; i < pad; i++) {
-        if (buf[--len] != pad) {
-            ERR_raise(ERR_LIB_PROV, PROV_R_BAD_DECRYPT);
-            return 0;
-        }
+    gp = ~(constant_time_is_zero_s(pad) | constant_time_lt_s(blocksize, pad));
+
+    for (i = 0; i < blocksize; i++) {
+        i_pad_lt = constant_time_lt(i, pad);
+        buf_pad_eq = constant_time_eq_s(buf[len - i - 1], pad);
```

Comment:
I think it would make sense to use either len OR blocksize throught this code as they are equal..

Reply:
Unresolved